records/2025-04-14.batch.saved.ses.receipts.md

```diff
@@ -0,0 +1,80 @@
+# Batch saved SES receipts
+
+Date: 2025-04-14
+
+## Status
+
+**DRAFT**.
+
+
+## Related ADRs
+
+N/A
+
+## Context
+
+During tests of the initial batch saving implementation in dev, we discovered that the "list" of receipts received by the ses callback lambda **always** contains a single receipt. This is a blocker to completing the batch saving implementation.
+
+Given we [appeared to be de-batching receipts](https://github.com/cds-snc/notification-lambdas/blob/main/sesemailcallbacks/ses_to_sqs_email_callbacks.py#L17) in our lambda code, the [initial implementation](https://github.com/cds-snc/notification-lambdas/blob/9c7ed0ba304513d9203b5a6af8e02e3cd768c612/sesemailcallbacks/ses_to_sqs_email_callbacks.py#L17) made minor adjustments to the lambda to batch up the receipts instead of sending them off one by one to the `eks-notification-canada-cadelivery-receipts` SQS queue for `process_ses_results` to pick up. Currently E2E testing this code is labourous and deterred validation of this assumption that was proved incorrect during testing.
```

Comment:
Interesting, do we know how many we might receive in a batch prior to the change? the chunk change seems like the right one though.

Reply:
Correct, anytime an event came into the lambda, the [`Records`](https://docs.aws.amazon.com/ses/latest/dg/receiving-email-action-lambda-event.html#receiving-email-action-lambda-event-lambdaaction) list in that event object always contains a single receipt. A bit of a blunder on my part failing to verify this assumption.